File: scala-hoots/src/main/scala/io/hoots/input/domain/WaveFileReaderDefaults.scala
package io.hoots.input.domain

import javax.sound.sampled.AudioFormat

/**
  * Created by rwadowski on 19.04.17.
  */
object WaveFileReaderDefaults {

  val sampleRate: Int = 44100
  val sampleSizeInBits: Int = 8
  val channels: Int = 1 //mono
  val signed: Boolean = true
  val bigEndian: Boolean = true
  val outputFormat = new AudioFormat(sampleRate, sampleSizeInBits, channels, signed, bigEndian)
  val BUFFER_SIZE: Int = 1024

  val decodedEncoding: AudioFormat.Encoding = AudioFormat.Encoding.PCM_SIGNED
  val decodedSampleSizeInBits: Int = 16
  val decodedBigEndian: Boolean = false

}

File: scala-hoots/src/main/java/io/hoots/signature/WaveFileReaderDefaults.java
package io.hoots.signature;

import javax.sound.sampled.AudioFormat;

/**
 * Created by rwadowski on 14.04.17.
 */
public class WaveFileReaderDefaults {

    public static final int sampleRate = 44100;
    public static final int sampleSizeInBits = 8;
    public static final int channels = 1; //mono
    public static final boolean signed = true;
    public static final boolean bigEndian = true;
    public static final AudioFormat outputFormat = new AudioFormat(sampleRate, sampleSizeInBits, channels, signed, bigEndian);
    public static final int BUFFER_SIZE = 1024;

    public static final AudioFormat.Encoding decodedEncoding = AudioFormat.Encoding.PCM_SIGNED;
    public static final int decodedSampleSizeInBits = 16;
    public static final boolean decodedBigEndian = false;

    private WaveFileReaderDefaults() {}
}